org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/serializer/SerializerScopeProvider.java

```diff
@@ -164,26 +164,25 @@ protected IScope getExecutableScope(XAbstractFeatureCall call, JvmIdentifiableEl
 			return new SingletonScope(EObjectDescription.create(operator, feature), IScope.NULLSCOPE);
 		}
 		if (call instanceof XAssignment) {
-			String shorthandName = PropertyUtil.tryGetShorthandName(simpleName);
```

Comment:
@szarnekow Is there a reason that we returned a scope only containing the shortened name here? The scope for setters would always have been `[foo]` instead of `[foo, setFoo]`. For getters, see below, both the exact and the shortened name are added to the scope.

Reply:
I would say `this.setFoo = 7;` should at least be a warning since this is not what I consider good coding style. Appears to be a bug in the scoping logic. From that POV we should include both variants here.

Reply:
@szarnekow This changes behaviour a little bit.
e.g. in [this](https://github.com/eclipse/xtext-xtend/blob/master/org.eclipse.xtend.ide.tests/longrunning/src/org/eclipse/xtend/ide/tests/refactoring/JavaRefactoringIntegrationTest.java#L537) test `setFoo` is refactored to `setBaz` which is supposed to conflict because `foo=null` would become `baz=null` which assigns to the field `baz` instead of calling the new refactored method `setBaz`.
By now adding `setBaz` to the scope we first try refactoring `foo=null` to `baz=null` and fail, then we try refactoring it to `setBaz=null`.

This is valid code but as you said not good coding style. Should we fix this somewhere else, or should we allow refactoring to potentially create new warnings?

Reply:
I'm ok (for now) if the refactoring produces new warnings. In the future I'd rather like to refactor the assignment to a feature call, e.g. instead of setBaz = null it should become setBaz(null)